packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -1039,29 +1224,58 @@ class _ActionsState extends State<Actions> {
     }
     listenedActions = widgetActions;
   }
+  
+  void _registerActions() => registry?.registerActions(registeredActionSubtreeKey);
+
+  void _unregisterActions() => registry?.unregisterActions(registeredActionSubtreeKey);
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    registry = Actions.maybeRegistryOf(context);
+    if (widget.registerActions) {
+  _registerActions();
```

Comment:
This will fail silently if no registry is in the tree. Is that expected or should it be an error to set registerActions to true if no registry can be found?

Reply:
Yeah, we should probably assert.

Reply:
OK, it asserts now.